File: C++/Day18.cpp
#include<bits/stdc++.h>
using namespace std;
class Palindrome {
    stack <char> s;
    queue <char> q;
    public :
    void pushCharacter(char ch){
        s.push(ch);
    }
    void enqueueCharacter(char ch){
        q.push(ch);
    }
    char popCharacter(){
        char c=s.top();
        s.pop();
        return c;
    }
    char dequeueCharacter(){
        char c = q.front();
        q.pop();
        return c;
    }
};

File: Python/Day18.py
class Palindrome:
    def __init__(self):
        self.stack=[]
        self.queue=[]
    def pushCharacter(self,ch):
        self.stack.append(ch)
    
    def enqueueCharacter(self,ch):
        self.queue.insert(0,ch)
     
    def popCharacter(self):
        return self.stack.pop()
    def dequeueCharacter(self):
        return self.queue.pop()
        
